postgres/database.go

```diff
@@ -35,11 +35,34 @@ func NewDB(ctx context.Context, connString string) (*DB, error) {
 // pass on unsanitised user input!
 func (db *DB) SwitchSchema(ctx context.Context, schema string) error {
 	slog.Info("Switching postgres schema", "schema", schema)
-	if _, err := db.Exec(ctx, fmt.Sprintf("BEGIN; SELECT pg_advisory_xact_lock(1); CREATE SCHEMA IF NOT EXISTS %s; COMMIT;", schema)); err != nil {
-		return fmt.Errorf("cannot create schema '%v': %w", schema, err)
+	_, err := db.Exec(
+		ctx,
+		fmt.Sprintf(
+			"BEGIN; SELECT pg_advisory_xact_lock(1); CREATE SCHEMA IF NOT EXISTS %s; SET search_path TO %s; COMMIT;",
+			schema,
+			schema,
+		),
+	)
+	if err != nil {
+		return fmt.Errorf("cannot create and switch to schema %q: %w", schema, err)
 	}
-	if _, err := db.Exec(ctx, fmt.Sprintf("SET search_path TO %s;", schema)); err != nil {
-		return fmt.Errorf("cannot set search_path to schema '%v': %w", schema, err)
+	return nil
+}
+
+// Set the search path to the specified value. If the search path contains non-existent schemas, this will error.
+// Beware that the schema here is not sanitised, as such this could be used to do SQL injection and should never
+// pass on unsanitised user input!
+func (db *DB) SetSearchPath(ctx context.Context, path string) error {
+	slog.Info("Changing postgres search path", "search_path", path)
+	_, err := db.Exec(
+		ctx,
+		fmt.Sprintf(
+			"BEGIN; SELECT pg_advisory_xact_lock(1); SET search_path TO %s; COMMIT;",
```

Comment:
Wat is het doel van de advisory lock hier?

Reply:
Ik heb het hier ook verwijderd, ik wist niet waarvoor die dienden dus had ze gewoon mee gekopieerd 😛